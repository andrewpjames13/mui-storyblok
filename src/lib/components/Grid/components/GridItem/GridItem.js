import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import Storyblok from '../../../../utils/Storyblok';

const GridItem = ({
  alignContent,
  alignItems,
  rootClass,
  direction,
  justify,
  lg,
  md,
  sm,
  wrap,
  spacing,
  xs,
  xl,
  sizeGrid,
  content,
  gridItemComponets,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass, { padding: '25px' });

  return (
    <Grid
      item
      container
      alignContent={alignContent}
      alignItems={alignItems}
      className={styles.root}
      direction={direction}
      justify={justify}
      lg={sizeGrid(lg)}
      md={sizeGrid(md)}
      sm={sizeGrid(sm)}
      wrap={wrap}
      spacing={Number(spacing)}
      xs={sizeGrid(xs)}
      xl={sizeGrid(xl)}
    >
      {/* {content.map((item, index) => createElement(
        gridItemComponets[item.component],
        Object.assign(item, { key: index }),
      ))} */}
    </Grid>
  );
};

export default GridItem;

GridItem.propTypes = {
  /**
   * developer prop to pass components to a gridItem
   */
  gridItemComponets: PropTypes.shape().isRequired,
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop: 'stretch'| 'center'| 'flex-start'| 'flex-end'| 'space-between'| 'space-around'
   * Defines the align-content style property. It's applied for all screen sizes.
   */
  alignContent: PropTypes.string,
  /**
   * mui prop: 'flex-start' | 'center' | 'flex-end' | 'stretch' | 'baseline'
   * Defines the align-items style property. It's applied for all screen sizes.
   */
  alignItems: PropTypes.string,
  /*
  * mui prop: 'row' | 'row-reverse' | 'column' | 'column-reverse'
  * Defines the flex-direction style property. It is applied for all screen sizes.
  */
  direction: PropTypes.string,
  /**
   * mui prop: 'flex-start' | 'center' | 'flex-end' | 'space-between' | 'space-around' | 'space-evenly'
   * Defines the justify-content style property. It is applied for all screen sizes.
   */
  justify: PropTypes.string,
  /**
   * mui prop: false | 'auto' | true | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
   * Defines the number of grids the component is going to use. It's applied for the breakpoint and wider screens if not overridden.
   */
  lg: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  /**
   * mui prop: false | 'auto' | true | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
   * Defines the number of grids the component is going to use. It's applied for the breakpoint and wider screens if not overridden.
   */
  md: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  /**
   * mui prop: false | 'auto' | true | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
   * Defines the number of grids the component is going to use. It's applied for the breakpoint and wider screens if not overridden.
   */
  sm: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  /**
   * mui prop:  0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10
   * Defines the space between the type item component. It can only be used on a type container component.
   */
  spacing: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
   * mui prop: 'nowrap' | 'wrap' | 'wrap-reverse'
   * Defines the flex-wrap style property. It's applied for all screen sizes.
   */
  wrap: PropTypes.string,
  /**
   * mui prop: false | 'auto' | true | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
   * Defines the number of grids the component is going to use. It's applied for the breakpoint and wider screens if not overridden.
   */
  xs: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  /**
   * mui prop: false | 'auto' | true | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
   * Defines the number of grids the component is going to use. It's applied for the breakpoint and wider screens if not overridden.
   */
  xl: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Content passed to MuiGrid to render
   * components: MuiButtonRedirect,
    MuiCard,
    MuiTypography,
    MuiMenu,
    MuiIcon,
    MuiExpansionPanel,
    MuiList
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,

  /** prop passed down from parent not in storyblok */
  sizeGrid: PropTypes.func.isRequired,
};

GridItem.defaultProps = {
  alignContent: 'stretch',
  alignItems: 'center',
  rootClass: [],
  direction: 'row',
  justify: 'center',
  lg: false,
  md: false,
  sm: false,
  spacing: '2',
  wrap: 'wrap',
  xl: false,
  xs: false,
};
