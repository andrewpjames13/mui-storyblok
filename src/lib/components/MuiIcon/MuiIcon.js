import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@material-ui/core/Icon';
import Storyblok from '../../utils/Storyblok';
import { muiStringProp } from '../../utils/customProps';

// import styles for icons in styleguide
import '../../styles/styles.scss';

/**
 * MuiIcon is used in storyblok redirect to react routes
 */

export const MuiIcon = ({
  color,
  fontSize,
  iconName,
  rootClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <Icon
      className={styles.root}
      color={color}
      fontSize={fontSize}
    >
      {iconName}
    </Icon>
  );
};

export default MuiIcon;

MuiIcon.propTypes = {
  /** any icon from https://material.io/resources/icons/?style=baseline */
  iconName: PropTypes.string.isRequired,
  /**
   * mui prop: "inherit", "primary", "secondary", "action", "error", "disabled"
   * The color of the component. It supports those theme colors that make sense for this component.
   * */
  color(props, propName, componentName) {
    const validProp = ['inherit', 'primary', 'secondary', 'action', 'error', 'disabled'];
    return muiStringProp(props, propName, componentName, validProp);
  },
  /**
   * mui prop: 'default', 'small', 'inherit', 'large'
   * The fontSize applied to the icon. Defaults to 24px, but can be configure to inherit font size.
   * */
  fontSize(props, propName, componentName) {
    const validProp = ['default', 'small', 'inherit', 'large'];
    return muiStringProp(props, propName, componentName, validProp);
  },
  /**
   * storyblok multiselect of css classes
   * Override or extend the styles applied to the component
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),
};

MuiIcon.defaultProps = {
  rootClass: [],
  color: 'secondary',
  fontSize: 'default',
};
