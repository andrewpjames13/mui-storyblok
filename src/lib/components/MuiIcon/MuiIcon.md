MuiIcon example:
[View All MUI Icons](https://material.io/resources/icons/?style=baseline)

```js
  <div styles={{ display: 'flex', flexDirection: 'row' }}>
    <MuiIcon
      iconName='home'
      color='primary'
    />
    <MuiIcon
      iconName='menu'
      color='secondary'
    />
    <MuiIcon
      iconName='close'
      color='action'
    />
    <MuiIcon
      iconName='accessibility'
      color='inherit'
    />
    <MuiIcon
      iconName='done'
      color='error'
    />

      <MuiIcon
      iconName='done'
      color='error'
      fontSize="small"
    />
  </div>
```

Social Icons for MuiIcon examples:
* iconName = 'facebook'
* iconName = 'bbb'
* iconName = 'google'
* iconName = 'linkedin'

```js
  <div styles={{ display: 'flex', flexDirection: 'row' }}>
    <MuiIcon
      iconName='facebook'
      color='primary'
      fontSize='large'
    />
    <MuiIcon
      iconName='bbb'
      color='secondary'
      fontSize='large'
    />
    <MuiIcon
      iconName='google'
      color='action'
      fontSize='large'
    />
    <MuiIcon
      iconName='linkedin'
      color='inherit'
      fontSize='large'
      rootClass={['margin-left: 20px']}
    />
  </div>
```