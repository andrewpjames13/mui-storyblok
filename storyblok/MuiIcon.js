module.exports = {
  name: 'MuiIcon',
  display_name: null,
  updated_at: '2020-03-31T18:27:02.209Z',
  schema: {
    iconName: {
      type: 'text',
      required: true,
      description: 'any icon from https://material.io/resources/icons/?style=baseline',
    },
    color: {
      type: 'option',
      use_uuid: true,
      options: [
        {
          value: 'inherit',
          name: 'inherit',
        },
        {
          value: 'primary',
          name: 'primary',
        },
        {
          value: 'secondary',
          name: 'secondary',
        },
        {
          value: 'action',
          name: 'action',
        },
        {
          value: 'error',
          name: 'error',
        },
        {
          value: 'disabled',
          name: 'disabled',
        },
      ],
      default_value: 'secondary',
      required: true,
    },
    fontSize: {
      type: 'option',
      use_uuid: true,
      options: [
        {
          value: 'default',
          name: 'default',
        },
        {
          value: 'small',
          name: 'small',
        },
        {
          value: 'inherit',
          name: 'inherit',
        },
        {
          value: 'large',
          name: 'large',
        },
      ],
      default_value: 'default',
    },
    rootClass: {
      type: 'options',
      min_options: '',
      options: [
        {
          value: 'color: white',
          name: 'color white',
        },
      ],
    },
  },
  image: null,
  preview_field: null,
  is_root: false,
  preview_tmpl: null,
  is_nestable: true,
  all_presets: [],
  preset_id: null,
  real_name: 'MuiIcon',
  component_group_uuid: null,
};
