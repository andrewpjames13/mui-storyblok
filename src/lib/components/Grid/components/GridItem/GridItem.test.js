import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiIcon from '../../../MuiIcon/MuiIcon';
import GridItem from './GridItem';

function setup() {
  const props = {
    sizeGrid: jest.fn(),
    gridItemComponents: {
      MuiIcon,
    },
    content: [{
      component: 'MuiIcon',
      iconName: 'android',
    }],
  };
  const comp = shallow(<GridItem {...props} />);
  return { comp, props };
}

describe('<GridItem />', () => {
  it('renders GridItem', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create((
      <MemoryRouter>
        <GridItem {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
