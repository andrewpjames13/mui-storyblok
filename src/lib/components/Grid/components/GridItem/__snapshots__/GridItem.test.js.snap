// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridItem /> snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-root-3 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
/>
`;
